python/cudf/cudf/testing/utils.py

```diff
@@ -0,0 +1,40 @@
+import random
+
+
+def _generate_rand_meta(obj, dtypes_list):
+    obj._current_params = {}
+    num_rows = obj._rand(obj._max_rows)
+    num_cols = obj._rand(obj._max_columns)
+
+    dtypes_meta = []
+
+    for _ in range(num_cols):
+        dtype = random.choice(dtypes_list)
+        null_frequency = random.uniform(0, 1)
+        cardinality = obj._rand(obj._max_rows)
+        meta = dict()
+        if dtype == "str":
+            # We want to operate near the limits of string column
+            # Hence creating a string column of size almost
+            # equal to 2 Billion bytes(sizeof(int))
+            meta["max_string_length"] = random.randrange(
+                0, 2000000000 / num_rows
```

Comment:
We may want to parameterize this number in case we change the string implementation later